src/app/qgisapp.cpp

```diff
@@ -7140,9 +7140,31 @@ void QgisApp::deleteSelected( QgsMapLayer *layer, QWidget *parent, bool promptCo
     return;
   }
   //display a warning
-  if ( promptConfirmation && QMessageBox::warning( parent, tr( "Delete features" ), tr( "Delete %n feature(s)?", "number of features to delete", numberOfSelectedFeatures ), QMessageBox::Ok | QMessageBox::Cancel ) == QMessageBox::Cancel )
+  if ( checkFeaturesVisible )
   {
-    return;
+    QgsFeature feat;
+    QgsFeatureIterator it = vlayer->getSelectedFeatures( QgsFeatureRequest().setSubsetOfAttributes( QgsAttributeList() ) );
```

Comment:
I think you'll need to also add a destination matching the canvas here

Reply:
What did you mean, to add destination CRS to the feature request? two lines below the canvas view rectangle is reprojected to layer CRS...

Reply:
Oops... Missed that!